src/osd/PrimaryLogPG.cc

```diff
@@ -837,47 +837,48 @@ bool PGLSPlainFilter::filter(const hobject_t &obj,
   return true;
 }
 
-bool PrimaryLogPG::pgls_filter(PGLSFilter *filter, hobject_t& sobj, bufferlist& outdata)
+bool PrimaryLogPG::pgls_filter(PGLSFilter& filter, hobject_t& sobj, bufferlist& outdata)
```

Comment:
i'd suggest extract this change out of this commit.

Reply:
also, i think another step is to mark the methods of `filter()` , `get_xattr()` and `reject_empty_xattr()` const, to assure that they do not have side-effect on `PGLSFilter`. and then we can safely move local variable`filter` in `do_pg_op()` into the loop.

Reply:
Extracted the key change from the refactoring and fixed the `const`-markings for `PGLSFilter`. I think the new commits will be useful also in the Ceph Class PR to address  [the `TODO` in `pgls_filter()`](https://github.com/ceph/ceph/pull/29651/files#diff-8a0759a622fe67c09d3bf6e5002ab77aR165).